src/test/java/com/amazon/ion/impl/IonReaderContinuableTopLevelBinaryTest.java

```diff
@@ -4436,4 +4448,190 @@ public void getAnnotationsAsStringFailsWhenTextIsUndefined(boolean constructFrom
+        assertSimpleStructCorrectlyParsed(false, inputBytes);
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void readMultipleNestedPrefixedStructs_1_1(boolean constructFromBytes) throws Exception {
+        byte[] input = hexStringToByteArray(cleanCommentedHexBytes(
+            "C8  | Struct Length = 8 \n" +
+            "17  | SID 11 \n" +
+            "C6  | Struct Length = 6 \n" +
+            "17  | SID 11 \n" +
+            "C4  | Struct Length = 4 \n" +
+            "17  | SID 11 \n" +
+            "C2  | Struct Length = 2 \n" +
+            "17  | SID 11 \n" +
+            "C0  | Struct Length = 0"
+        ));
+        reader = readerForIon11(input, constructFromBytes);
+        assertSequence(
+            container(IonType.STRUCT,
+                container(IonType.STRUCT,
+                    container(IonType.STRUCT)
+                )
+            ),
+            next(null)
+        );
+        closeAndCount();
+    }
+
+    /**
+     * Checks that the reader correctly reads symbol zero, which requires a special FlexSym, as a field name.
+     */
+    private void assertStructWithSymbolZeroFieldNamesCorrectlyParsed(boolean constructFromBytes, String inputBytes) throws Exception {
+        reader = readerForIon11(hexStringToByteArray(cleanCommentedHexBytes(inputBytes)), constructFromBytes);
+        assertSequence(
+            container(IonType.STRUCT,
+                next(IonType.BOOL), fieldNameSymbolZero(), booleanValue(true),
+                next(IonType.BOOL), fieldNameSymbolZero(), booleanValue(true),
+                next(IonType.BOOL), fieldName("name"), booleanValue(true),
+                next(IonType.BOOL), fieldNameSymbolZero(), booleanValue(true)
+            ),
+            next(null)
+        );
+        closeAndCount();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+        // SID 0 in fixed-length SID struct
+        "C8      | Struct Length = 8 \n" +
+        "01      | SID 0 \n" +
+        "5E      | true \n" +
+        "01      | FlexSym SID 0 \n" +
+        "5E      | true \n" +
+        "09      | FlexSym SID 4 (name) \n" +
+        "5E      | true \n" +
+        "01      | FlexSym SID 0 \n" +
+        "5E      | true",
+        // SID 0 in variable-length FlexSym struct
+        "FD      | Variable length SID struct \n" +
+        "17      | Length = FlexUInt 11 \n" +
+        "01 90   | SID 0 \n" +
+        "5E      | true \n" +
+        "01 90   | FlexSym SID 0 \n" +
+        "5E      | true \n" +
+        "09      | FlexSym SID 4 (name) \n" +
+        "5E      | true \n" +
+        "01 90   | FlexSym SID 0 \n" +
+        "5E      | true",
+        // SID 0 before and after transitioning from SIDs to FlexSyms
+        "FC      | Variable length SID struct \n" +
+        "17      | Length = FlexUInt 11 \n" +
+        "01      | SID 0 \n" +
+        "5E      | true \n" +
+        "00      | switch to FlexSym encoding \n" +
```

Comment:
Should be _FlexUInt 0_ (`01`) to switch to FlexSym encoding.

Reply:
I copied test data from here: https://github.com/amazon-ion/ion-java/pull/665/files#diff-c491f83b80fd84cfed9e62a411f61789c10d9731d4afbb7fefe538738dd3cd27R557

Do we have a bug in both places? Is there a reason why we can't use `00`?

Reply:
> Do we have a bug in both places?

It looks like there is a bug in both places.

>  Is there a reason why we can't use 00?

`00` isn't a valid FlexUInt. The fact that there's no 1s in that byte means that the FlexUInt is at least another 9 byts.

Reply:
> Do we have a bug in both places? Is there a reason why we can't use 00?

I guess it limits the range of field name SIDs... but I might be happy with that.